lib/install/management/upgrade.go

```diff
@@ -99,10 +101,37 @@ func (d *Dispatcher) Upgrade(vch *vm.VirtualMachine, conf *config.VirtualContain
 
 	d.deleteUpgradeImages(ds, settings)
 	d.retryDeleteSnapshot(snapshotName, conf.Name)
+
 	// return the error message for upgrade
 	return err
 }
 
+func (d *Dispatcher) Rollback(vch *vm.VirtualMachine, conf *config.VirtualContainerHostConfigSpec, settings *data.InstallerData) error {
+
+	// some setup that is only necessary because we didn't just create a VCH in this case
+	d.appliance = vch
+	d.setDockerPort(conf, settings)
+	if !strings.HasSuffix(conf.Target, "/sdk") {
+		conf.Target = fmt.Sprintf("%s/sdk", conf.Target)
```

Comment:
I don't know. You would think that `conf.Target` would have the `/sdk` suffix, but it does not, and it is needed.